### PR TITLE
React-Native 0.60

### PR DESCRIPTION
**Pull request by Fausto95**
Bumping RN's version to 0.60.5(latest as of 9/20/2019).

Changes:
 - [x] Androidx Support
 - [x] Auto Linking
 - [x] Hermes Engine

### CODE REVIEW

**Comment by Fausto95** [comment]
This will only be merged after we fix all the lags in the app

File: ios/Podfile [hunk 1]
```diff
@@ -10,18 +10,23 @@ target 'Coorpacademy' do
...
-    'DevSupport', # Include this to enable In-App Devmenu if RN >= 0.43
-    'RCTText',
-    'RCTNetwork',
-    'RCTWebSocket', # Needed for debugging
-    'RCTAnimation', # Needed for FlatList and animations running on native UI thread
-    # Add any other subspecs you want to use in your project
-    'RCTImage', # Needed for react-native-svg
-    'RCTLinkingIOS', # Needed for react-navigation
-  ]
+  pod 'React', :path => "#{rn_path}/"
```
Comment (Fausto95) [inline ios/Podfile]: I think that array way is most fashion than this one

File: jest.config.unit.js [hunk 2]
```diff
@@ -15,6 +15,7 @@ module.exports = {
   reporters: ['default', 'jest-junit'],
   collectCoverage: true,
   coverageReporters: ['lcov', 'text', 'html'],
+  testEnvironment: 'jsdom',
```
Comment (Fausto95) [inline jest.config.unit.js]: https://github.com/facebook/jest/issues/4359
https://github.com/facebook/jest/issues/6434

File: package.json [hunk 3]
```diff
@@ -24,6 +24,7 @@
     "start:android": "react-native run-android",
     "start:android:release": "npm run start:android -- --variant=release",
     "prebuild:ios:cmd": "npm run -s xcode:check",
+    "prebuild:android:cmd": "npx jetify",
```
Comment (Fausto95) [inline package.json]: https://github.com/mikehardy/jetifier

Comment (Fausto95) [reply package.json]: Also, the cmd `react-native run-android` does this by default(under the hood), but we're adding it in order to build the app when using fastlane as it does not run the mentioned command

File: ios/Podfile [hunk 4]
```diff
@@ -10,18 +10,23 @@ target 'Coorpacademy' do
...
+  pod 'React-DevSupport', :path => "#{rn_path}/React"
+  pod 'React-RCTActionSheet', :path => "#{rn_path}/Libraries/ActionSheetIOS"
+  pod 'React-RCTAnimation', :path => "#{rn_path}/Libraries/NativeAnimation"
+  pod 'React-RCTBlob', :path => "#{rn_path}/Libraries/Blob"
+  pod 'React-RCTImage', :path => "#{rn_path}/Libraries/Image"
+  pod 'React-RCTLinking', :path => "#{rn_path}/Libraries/LinkingIOS"
+  pod 'React-RCTNetwork', :path => "#{rn_path}/Libraries/Network"
+  pod 'React-RCTSettings', :path => "#{rn_path}/Libraries/Settings"
+  pod 'React-RCTText', :path => "#{rn_path}/Libraries/Text"
+  pod 'React-RCTVibration', :path => "#{rn_path}/Libraries/Vibration"
```
Comment (Fausto95) [inline ios/Podfile]: Probably not being used

File: ios/Podfile [hunk 5]
```diff
@@ -10,18 +10,23 @@ target 'Coorpacademy' do
...
+  pod 'React', :path => "#{rn_path}/"
+  pod 'React-Core', :path => "#{rn_path}/React"
+  pod 'React-DevSupport', :path => "#{rn_path}/React"
+  pod 'React-RCTActionSheet', :path => "#{rn_path}/Libraries/ActionSheetIOS"
+  pod 'React-RCTAnimation', :path => "#{rn_path}/Libraries/NativeAnimation"
+  pod 'React-RCTBlob', :path => "#{rn_path}/Libraries/Blob"
+  pod 'React-RCTImage', :path => "#{rn_path}/Libraries/Image"
+  pod 'React-RCTLinking', :path => "#{rn_path}/Libraries/LinkingIOS"
+  pod 'React-RCTNetwork', :path => "#{rn_path}/Libraries/Network"
+  pod 'React-RCTSettings', :path => "#{rn_path}/Libraries/Settings"
```
Comment (Fausto95) [inline ios/Podfile]: Probably not being used

File: ios/Podfile [hunk 6]
```diff
@@ -10,18 +10,23 @@ target 'Coorpacademy' do
...
-    'RCTWebSocket', # Needed for debugging
-    'RCTAnimation', # Needed for FlatList and animations running on native UI thread
-    # Add any other subspecs you want to use in your project
-    'RCTImage', # Needed for react-native-svg
-    'RCTLinkingIOS', # Needed for react-navigation
-  ]
+  pod 'React', :path => "#{rn_path}/"
+  pod 'React-Core', :path => "#{rn_path}/React"
+  pod 'React-DevSupport', :path => "#{rn_path}/React"
+  pod 'React-RCTActionSheet', :path => "#{rn_path}/Libraries/ActionSheetIOS"
```
Comment (Fausto95) [inline ios/Podfile]: Probably not being used

File: ios/Podfile (hunk 4 above)
Comment (Fausto95) [reply ios/Podfile]: Ok just figured out that these are the defaults one
https://react-native-community.github.io/upgrade-helper/?from=0.59.9&to=0.60.5

File: ios/Podfile (hunk 5 above)
Comment (Fausto95) [reply ios/Podfile]: Ok just figured out that these are the defaults one
https://react-native-community.github.io/upgrade-helper/?from=0.59.9&to=0.60.5

File: ios/Podfile (hunk 6 above)
Comment (Fausto95) [reply ios/Podfile]: Ok just figured out that these are the defaults one
https://react-native-community.github.io/upgrade-helper/?from=0.59.9&to=0.60.5

**Comment by Fausto95** [comment]
`MUST FUNCTIONALLY TEST IT ON A REAL ANDROID DEVICE`

File: android/settings.gradle [hunk 7]
```diff
@@ -1,4 +1,7 @@
 rootProject.name = 'Coorpacademy'
+
+apply from: file("../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesSettingsGradle(settings)
```
Comment (VincentCATILLON) [inline android/settings.gradle]: this file is for autolinking, we should so, remove all deps added by `react-native link` before

File: package.json [hunk 8]
```diff
@@ -102,25 +103,25 @@
...
     "react-native-snap-carousel": "^3.8.0",
-    "react-native-sound": "^0.10.12",
+    "react-native-sound": "0.11.0",
     "react-native-splash-screen": "^3.2.0",
     "react-native-status-bar-height": "^2.3.1",
     "react-native-svg": "^9.4.0",
-    "react-native-video": "4.4.1",
-    "react-native-webview": "^5.11.0",
+    "react-native-video": "5.0.0",
+    "react-native-webview": "^6.9.0",
```
Comment (VincentCATILLON) [inline package.json]: why these dependencies have been updated ? same issue than going to RN 0.59 that introduced translations issues etc.

File: android/settings.gradle (hunk 7 above)
Comment (Fausto95) [reply android/settings.gradle]: This is not it, Auto-linking adds libs deps here, so should not remove this

File: package.json (hunk 8 above)
Comment (Fausto95) [reply package.json]: Cuz they need to be updated
There's no breaking changes, check it out and test

File: .flowconfig [hunk 9]
```diff
@@ -21,6 +21,9 @@
 .*/Libraries/react-native/React.js
 .*/Libraries/react-native/ReactNative.js
 
+; Flow doesn't support platforms
```
Comment (VincentCATILLON) [inline .flowconfig]: some flow config have been forgotten: https://react-native-community.github.io/upgrade-helper/?from=0.59.9&to=0.60.5

Comment (VincentCATILLON) [reply .flowconfig]: you also have to bump flow version

File: android/app/src/main/java/com/coorpacademy/app/MainApplication.java [hunk 10]
```diff
@@ -39,6 +43,7 @@ public boolean getUseDeveloperSupport() {
       return BuildConfig.DEBUG;
     }
 
+
```
Comment (VincentCATILLON) [inline android/app/src/main/java/com/coorpacademy/app/MainApplication.java]: Please update this file to follow the original file: https://raw.githubusercontent.com/react-native-community/rn-diff-purge/release/0.60.5/RnDiffApp/android/app/src/main/java/com/rndiffapp/MainApplication.java

File: ios/Podfile (hunk 1 above)
Comment (VincentCATILLON) [inline ios/Podfile]: this file is not up to date with one in RN repository, it should include some things like `require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'`

File: ios/Podfile [hunk 11]
```diff
@@ -10,18 +10,23 @@ target 'Coorpacademy' do
...
+  pod 'React-RCTLinking', :path => "#{rn_path}/Libraries/LinkingIOS"
+  pod 'React-RCTNetwork', :path => "#{rn_path}/Libraries/Network"
+  pod 'React-RCTSettings', :path => "#{rn_path}/Libraries/Settings"
+  pod 'React-RCTText', :path => "#{rn_path}/Libraries/Text"
+  pod 'React-RCTVibration', :path => "#{rn_path}/Libraries/Vibration"
+  pod 'React-RCTWebSocket', :path => "#{rn_path}/Libraries/WebSocket"
+  pod 'React-cxxreact', :path => "#{rn_path}/ReactCommon/cxxreact"
+  pod 'React-jsi', :path => "#{rn_path}/ReactCommon/jsi"
+  pod 'React-jsiexecutor', :path => "#{rn_path}/ReactCommon/jsiexecutor"
+  pod 'React-jsinspector', :path => "#{rn_path}/ReactCommon/jsinspector"
```
Comment (VincentCATILLON) [inline ios/Podfile]: Looking for this, dunno if it's a good thing to have deps there, they should be manage by `react-native.config.js` file that list all native deps included in project

File: android/settings.gradle (hunk 7 above)
Comment (VincentCATILLON) [reply android/settings.gradle]: https://github.com/react-native-community/cli/blob/master/docs/commands.md#config

Comment (VincentCATILLON) [reply android/settings.gradle]: so they should be removed from `settings.gradle` and `MainApplication.java` (or autolinking is useless lol)

Comment (VincentCATILLON) [reply android/settings.gradle]: for supported libs:

https://github.com/react-native-community/cli/blob/master/docs/autolinking.md#how-can-i-autolink-a-local-library

for unsupported libs:

https://github.com/react-native-community/cli/blob/master/docs/autolinking.md#how-can-i-disable-autolinking-for-unsupported-library

File: .flowconfig (hunk 9 above)
Comment (Fausto95) [reply .flowconfig]: I did not forget, we dont need to upgrade flow

**Comment by VincentCATILLON** [comment]
At this time, too much files are not up to date with RN project folder

We should init a new project with RN 0.60.x and restore our files, and (the most) deps we have in the same version (or bumped if strictly necessary), to have exactly the same files than RN project has (we did it for 0.59.x in 2nd time and was the most accurate way to go to a clean project).

**Comment by VincentCATILLON** [comment]
See Trello ticket, standby because author don't want to spend any other time on it

**Pull request closed without merging**